src/snowflake/snowpark/_internal/data_source/drivers/oracledb_driver.py

```diff
@@ -117,6 +118,43 @@ def prepare_connection(
             conn.outputtypehandler = output_type_handler
         return conn
 
+    def udtf_class_builder(self, fetch_size: int = 1000) -> type:
+        create_connection = self.create_connection
+
+        def udtf_handler(cursor, metadata):
+            from oracledb import (
+                DB_TYPE_CLOB,
+                DB_TYPE_NCLOB,
+                DB_TYPE_LONG,
+                DB_TYPE_BLOB,
+                DB_TYPE_RAW,
+                DB_TYPE_LONG_RAW,
+            )
+
+            def convert_to_hex(value):
+                return value.hex() if value is not None else None
+
+            if metadata.type_code in (DB_TYPE_CLOB, DB_TYPE_NCLOB):
+                return cursor.var(DB_TYPE_LONG, arraysize=cursor.arraysize)
+            elif metadata.type_code in (DB_TYPE_BLOB, DB_TYPE_RAW, DB_TYPE_LONG_RAW):
+                return cursor.var(
+                    DB_TYPE_RAW, arraysize=cursor.arraysize, outconverter=convert_to_hex
+                )
```

Comment:
curious why we need this `convert_to_hex` in the udtf approach but not in the parquet file ingestion?

Reply:
in parquet ingestion, convert to hex is covered in data_source_data_to_pandas_df(), while we don't have a pandas df intermedia state in udtf ingestion, we need to do it here.
in theory, the client ingestion could also move convert to hex here to improve performance

Reply:
if that's the case, can you create a follow up to align the implementation between the two?

Reply:
do you mind adding the JIRA as a comment so that we do not forget?